hail/python/hail/methods/statgen.py

```diff
@@ -1758,16 +1758,25 @@ def split_rows(expr, rekey):
         if isinstance(ds, MatrixTable):
             mt = (ds.annotate_rows(**{new_id: expr})
                   .explode_rows(new_id))
+            if rekey:
+                mt = mt.key_rows_by()
+            else:
```

Comment:
I see this is the same structure as the code for Table below. I assume the reason for having two key-by operations (before and after the annotate) has to do with split_multi.

Reply:
`rekey` is controlling whether we know things will be sorted correctly after exploding. If we do (`rekey=false`) then we shorten the keys to 'locus', because the locus field isn't changing, so the partition bounds truncated to locus will still be valid, then we annotate, then extend the key back to `['locus', 'alleles']` with the flag that says it's already sorted. The last step is basically a no-op, the result is an RVD keyed by `['locus', 'alleles']`, but where all the partition bounds only have locus information.

If the keys might not be in order after the explode, then there's nothing to do but shuffle after, which is the `rekey=true` path.